data-import/desktop-csv-import/desktop-csv-import.adoc

```diff
@@ -0,0 +1,420 @@
+
+.Orders.csv
+The values in the orders.csv (per the column names) are for orderID, orderDate, and shipCountry.
+Again, we can evaluate the values and determine any conversions to apply.
+
+OrderID looks like it is an integer, so we can convert that using the `toInteger()` function.
+The orderDate column is certainly in a date format and will require us to format it using the `datetime()` function.
+Finally, the shipCountry values are characters, so we can leave that as a string.
+
+[NOTE]
+--
+If you are using a version of Neo4j prior to 3.4, the `datetime()` function will not be available.
+That function was released in 3.4 and is supported from that version forward.
+--
+
+Just as we did with the last CSV file, let us look at the results of these conversions without importing the data.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/orders.csv' AS row
+RETURN toInteger(row.orderID), datetime(replace(row.orderDate,' ','T')), row.shipCountry
+LIMIT 5
+----
+
+image::{img}desktop_import_orders_convert.jpg[link="{img}desktop_import_orders_convert.jpg",role="popup-link"]
+
+There was one tricky thing with this CSV in the `orderDate` column.
+Neo4j's datetime format will accept many formats, but the time value must start with a `T`.
+The CSV file did not have the 'T' joining the date and time values but had a space between them instead.
+We used the `replace()` function to change the space to the character 'T' and get the string into the expected format.
+Then, we wrapped the `datetime()` function around that to convert the changed string to a datetime value.
+
+.Order-details.csv
+The values in the order-details.csv (from column names) are for productID, orderID, and quantity.
+Let us look at which ones will need to be converted.
+
+Our product ID is also from our products.csv file, where we converted that value to an integer.
+We will do the same here to ensure we match formats.
+The order ID field contains values from our orders.csv file, so we will match our previous conversion and translate this field to an integer, as well.
+The quantity field in this file is a numeric value.
+We can convert this to an integer with the `toInteger()` function we have been using.
+
+The results of these conversions are in the code below.
+Remember that we still are not loading any data yet.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/order-details.csv' AS row
+RETURN toInteger(row.productID), toInteger(row.orderID), toInteger(row.quantity)
+LIMIT 8
+----
+
+image::{img}desktop_import_details_convert.jpg[link="{img}desktop_import_details_convert.jpg",role="popup-link"]
+
+== Loading the data!
+
+Now that we have determined that the CSV file data looks ok, and we have verified how `LOAD CSV` sees the data and converted any non-string values, we are almost ready to create the data in our graph database!
+To do that, we use Cypher statements alongside the `LOAD CSV` commands we used above.
```

Comment:
Cypher update statements

Reply:
We haven't loaded the data yet, though. What are we updating?